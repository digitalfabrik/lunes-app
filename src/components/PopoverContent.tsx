import React from 'react'
import { View, Text, StyleSheet } from 'react-native'
import { InfoIcon } from '../../assets/images'
import { widthPercentageToDP as wp } from 'react-native-responsive-screen'
import { COLORS } from '../constants/colors'

export const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: COLORS.lunesBlack,
    width: wp('80%'),
    height: 60,
    paddingVertical: 8,
    paddingHorizontal: 9,
    borderRadius: 2
  },
  message: {
    color: COLORS.lunesWhite,
    fontSize: wp('3.5%'),
    fontWeight: 'normal',
    fontFamily: 'SourceSansPro-Regular',
    width: wp('60%'),
    marginLeft: 8
  }
})
const PopoverContent = () => (
  <View style={styles.container}>
    <InfoIcon width={30} height={30} />
    <Text style={styles.message}>Ups, you forgot someting. You have to add an article to continue.</Text>
  </View>
)

export default PopoverContent
