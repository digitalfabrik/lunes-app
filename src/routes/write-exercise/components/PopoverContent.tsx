import React from 'react'
import { InfoIcon } from '../../../../assets/images'
import { widthPercentageToDP as wp } from 'react-native-responsive-screen'
import { COLORS } from '../../../constants/colors'
import labels from '../../../constants/labels.json'
import styled from 'styled-components/native'

const StyledComponent = styled.View`
  flex-direction: row;
  align-items: center;
  background-color: ${COLORS.lunesBlack};
  width: ${wp('80%')};
  height: 60;
  padding-right: 8;
  padding-left: 8;
  padding-top: 9;
  padding-right: 9;
  border-radius: 2;
`
const StyledMessage = styled.Text`
  color: ${COLORS.lunesWhite};
  font-size: ${wp('3.5%')};
  font-weight: normal;
  font-family: SourceSansPro-Regular;
  width: ${wp('60%')};
  margin-left: 8;
`

const PopoverContent = () => (
  <StyledComponent>
    <InfoIcon width={30} height={30} />
    <StyledMessage>{labels.exercises.write.feedback.articleMissing}</StyledMessage>
  </StyledComponent>
)

export default PopoverContent
