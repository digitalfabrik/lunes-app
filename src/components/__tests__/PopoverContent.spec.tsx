import 'react-native'
import React from 'react'
import PopoverContent from '../PopoverContent'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

describe('components', () => {
  describe('PopoverContent', () => {
    it('renders correctly across screens', () => {
      const component = shallow(<PopoverContent />)
      expect(toJson(component)).toMatchSnapshot()
    })
  })
})
