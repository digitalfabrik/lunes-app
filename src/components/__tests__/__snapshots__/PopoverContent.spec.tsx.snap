// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components PopoverContent renders correctly across screens 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#000e38",
      "borderRadius": 2,
      "flexDirection": "row",
      "height": 60,
      "paddingHorizontal": 9,
      "paddingVertical": 8,
      "width": 600,
    }
  }
>
  <[object Object]
    height={30}
    width={30}
  />
  <Text
    style={
      Object {
        "color": "#f9fafb",
        "fontFamily": "SourceSansPro-Regular",
        "fontSize": 26.5,
        "fontWeight": "normal",
        "marginLeft": 8,
        "width": 450,
      }
    }
  >
    Ups, you forgot someting. You have to add an article to continue.
  </Text>
</View>
`;
